lib/autocomplete-manager.coffee

```diff
@@ -1,183 +1,111 @@
-    keys =
-      "escape": "autocomplete-plus:cancel"
-
-    completionKey = atom.config.get("autocomplete-plus.confirmCompletion") || ''
-    navigationKey = atom.config.get("autocomplete-plus.navigateCompletions") || ''
-
-
-    keys['tab'] = "autocomplete-plus:confirm" if completionKey.indexOf('tab') > -1
-    keys['enter'] = "autocomplete-plus:confirm" if completionKey.indexOf('enter') > -1
-
-    if @items?.length > 1 and navigationKey == "up,down"
-      keys['up'] =  "autocomplete-plus:select-previous"
-      keys['down'] = "autocomplete-plus:select-next"
-    else
-      keys["ctrl-n"] = "autocomplete-plus:select-next"
-      keys["ctrl-p"] = "autocomplete-plus:select-previous"
-
-    @keymaps = atom.keymaps.add(
-      'AutocompleteManager',
-      'atom-text-editor:not(.mini) .autocomplete-plus': keys
-    )
-
-    @compositeDisposable.add @keymaps
-
-  removeKeyboardInteraction: ->
-    @keymaps?.dispose()
-    @compositeDisposable.remove(@keymaps)
+    @handleCommands()
+    @fuzzyProvider = new FuzzyProvider()
 
   updateCurrentEditor: (currentPaneItem) =>
-    @cancel() unless currentPaneItem == @editor
-
-  confirmSelection: =>
-    @emitter.emit 'do-confirm-selection'
-
-  onDoConfirmSelection: (cb) ->
-    @emitter.on 'do-confirm-selection', cb
+    return unless currentPaneItem?
+    return if currentPaneItem is @editor
 
-  selectNext: =>
-    @emitter.emit 'do-select-next'
+    # Stop listening to buffer events
+    @bufferSavedSubscription?.dispose()
+    @bufferChangedSubscription?.dispose()
 
-  onDoSelectNext: (cb) ->
-    @emitter.on 'do-select-next', cb
+    # Stop listening to editor events
+    @editorCursorMovedSubscription?.dispose()
 
-  selectPrevious: =>
-    @emitter.emit 'do-select-previous'
+    # Disqualify invalid pane items
+    # TODO
+    if false
+      @editor = null
+      @editorView = null
+      @buffer = null
 
-  onDoSelectPrevious: (cb) ->
-    @emitter.on 'do-select-previous', cb
-
-  # Private: Checks whether the current file is blacklisted
-  #
-  # Returns {Boolean} that defines whether the current file is blacklisted
-  currentFileBlacklisted: ->
-    blacklist = (atom.config.get("autocomplete-plus.fileBlacklist") or "")
-      .split ","
-      .map (s) -> s.trim()
-
-    fileName = path.basename @editor.getBuffer().getPath()
-    for blacklistGlob in blacklist
-      if minimatch fileName, blacklistGlob
-        return true
+    # Track the new editor, editorView, and buffer
+    @editor = currentPaneItem
+    @editorView = atom.views.getView(@editor)
+    @buffer = @editor.getBuffer()
 
-    return false
+    # Subscribe to buffer events:
+    @bufferSavedSubscription = @buffer.onDidSave(@bufferSaved)
+    @bufferChangedSubscription = @buffer.onDidChange(@bufferChanged)
 
+    # Subscribe to editor events:
+    # Close the overlay when the cursor moved without changing any text
+    @editorCursorMovedSubscription = @editor.onDidChangeCursorPosition(@cursorMoved)
```

Comment:
Same explanation as the buffer subscription above. A single `AutocompleteManager` tracks all editors / buffers (but only one of each at a time) for the lifetime of the package's activation.